doc/requirements/sdid.rst

```diff
@@ -0,0 +1,261 @@
+component :math:`i` is :math:`N_i V_i`. Thus, to conserve this number of atoms over a
+total volume :math:`V_b = \sum_i V_i`, we calculate the homogenized number density :math:`\bar{N_i}`
+of component :math:`i` as
+
+.. math::
+    :label: conserveAtoms
+		N_i V_i = \bar{N_i} V_b \\
+		\bar{N_i} = \frac{N_i V_i}{V_b}
+Thus, homogenized number densities are equal to heterogeneous number densities multiplied by the component volume
+fraction.
+
+This calculation is performed in :py:meth:`armi.reactor.composites.ArmiObject.getNumberDensity`.
+
+.. figure:: /.static/block_homogenization.png
+    :align: center
+
+    **Figure 1.** Homogenizing pins, duct, wire, cladding, and coolant into a uniform block
+
+Similarly, :py:meth:`~armi.reactor.composites.ArmiObject.getMass` can get the mass of some or all
+nuclides in a structure and :py:meth:`~armi.reactor.composites.ArmiObject.getNumberOfAtoms` can get the number
+of atoms.
+
+Calculation Of Volume Fractions
+"""""""""""""""""""""""""""""""
+To support the homogenization responsibility, the ``reactor`` package is responsible for computing the volume fractions
+:math:`v_i` of each component. Generally, ``components`` are responsible for computing their own volume :math:`V_i`, and
+other levels of the hierarchy simply have to evaluate the simple formula,
+
+.. math::
+    :label: areaFraction
+		v_i = \frac{V_i}{\sum_j V_j}
+.. WARNING::
+	Often, components only compute their area and their height is inherited as the height of the
+ 	containing block. There are exceptions for more complex geometries.
+
+For user convenience, the dimensions of one component may be left undefined in input. If one and only one
+component has undefined area, then the block will compute the area automatically. This is useful, for example,
+when a complex shape exists for the coolant material between all pins. In this scenario, the maximum block
+area is computed using the largest pitch :math:`p_{max}` (generally the interstitial gap). For hex geometry, the missing area :math:`A_{missing}`
+is computed as:
+
+.. math::
+    :label: missingArea
+		A_{missing} = p_{max}^2 \frac{\sqrt{3}}{2} - \sum_{i \neq missing}{A_i}
+
+Hot and input dimensions
+^^^^^^^^^^^^^^^^^^^^^^^^
+ARMI treats dimensions and material properties as functions of temperature. However, a pure
+physical analogy is challenging for several reasons. These reasons and
+the implementation details are explained here.
+
+For a typical ``component``, users may define most dimensions at any temperature they desire
+(the *Input temperature*), as explained in :doc:`/user/inputs/composition_file`. These
+dimensions will be thermally-expanded up to the *Hot temperature* as input. For most shapes and
+components, this works as expected, however:
+
+* In Hex geometries, the outer hexagonal boundary is currently limited to be consistent across
+  all objects in a core. This stems from some physics solver requirements of structured meshes.
```

Comment:
"all objects" is not correct, since an "object" could be an assembly, a block, a component. I think this should say "all assemblies"